client/db/src/storage_cache.rs

```diff
@@ -302,12 +313,14 @@ pub struct CachingState<S: StateBackend<HasherFor<B>>, B: BlockT> {
 	/// Backing state.
 	state: S,
 	/// Cache data.
-	pub cache: CacheChanges<B>,
+	cache: Option<CacheChanges<B>>,
+	/// Data associated to the backend that created this instance.
+	backend: Option<CachingStateBackend<B>>,
```

Comment:
Do `cache ` and `backend `really need to be `Option`? It look slike `drop` implementation does not really require that.